File: 128 Longest Consecutive Sequence/128 Longest Consecutive Sequence.swift
class Solution {
  func longestConsecutive(_ nums: [Int]) -> Int {
    guard nums.count != 0 else {
      return 0
    }

    var longestConsecutiveCount: Int = 1
    let numsSet: Set<Int> = Set(nums)

    for num in numsSet {
      if numsSet.contains(num - 1) {
        continue
      }

      var currentLongestConsecutiveCount: Int = 1

      while numsSet.contains(num + currentLongestConsecutiveCount) {
        currentLongestConsecutiveCount += 1
      }

      longestConsecutiveCount = max(longestConsecutiveCount, currentLongestConsecutiveCount)
    }

    return longestConsecutiveCount
  }
}

File: 128 Longest Consecutive Sequence/128 Longest Consecutive Sequence.ts
function longestConsecutive(nums: number[]): number {
  if (nums.length === 0) return 0;

  let longestConsecutiveLength: number = 1;
  const numsSet: Set<number> = new Set(nums);

  for (const num of numsSet) {
    if (numsSet.has(num - 1)) continue;

    let currentLongestConsecutiveLength: number = 1;

    while (numsSet.has(num + currentLongestConsecutiveLength)) {
      currentLongestConsecutiveLength++;
    }

    longestConsecutiveLength = Math.max(
      longestConsecutiveLength,
      currentLongestConsecutiveLength
    );
  }

  return longestConsecutiveLength;
}
